salt/_auth/kubernetes_rbac.py

```diff
@@ -1,31 +1,152 @@
 import base64
 import logging
 
+MISSING_DEPS = []
+
+try:
+    import kubernetes.client
+    import kubernetes.config
+except ImportError:
+    MISSING_DEPS.append('kubernetes')
+
+try:
+    import requests
+except ImportError:
+    MISSING_DEPS.append('requests')
+
+
 log = logging.getLogger(__name__)
 
 __virtualname__ = 'kubernetes_rbac'
 
 def __virtual__():
-    return __virtualname__
-
-
-ADMIN_USERNAME = b'admin'
-ADMIN_PASSWORD = b'admin'
-ADMIN_TOKEN = base64.encodestring(
-    b'{}:{}'.format(ADMIN_USERNAME, ADMIN_PASSWORD)).rstrip()
-AUTH_HANDLERS = {
-    'basic': {
-        'auth': lambda username, token: all([
-            username == ADMIN_USERNAME,
-            token == ADMIN_TOKEN,
-        ]),
-        'groups': lambda _u, _t: [
-            'node_admin',
-        ],
-    },
+    if MISSING_DEPS:
+        return False, 'Missing Python dependencies: {}'.format(
+            ', '.join(MISSING_DEPS))
+    else:
+        return __virtualname__
+
+
+AUTH_HANDLERS = {}
+
+
+def _log_exceptions(f):
+    def wrapped(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except:
+            log.exception('Exception thrown')
+            raise
+
+    return wrapped
+
+
+@_log_exceptions
+def _auth_basic(kubeconfig, username, token):
+    decoded = base64.decodestring(token)
+    if ':' not in decoded:
+        log.warning('Invalid Basic token format: missing ":"')
+        return False
+
+    (token_username, _) = decoded.split(':', 1)
```

Comment:
minor: No need for the parentheses on the left-hand side, and even more explicit to use indexing:

```suggestion
    token_username = decoded.split(':', 1)[0]
```

Reply:
Agree the parentheses are superfluous. I prefer `a, _ = foo.split(b, 1)` over `a = foo.split(b, 1)[0]`, because
- it makes it clearer there's only a single split about to happen
- if there's no split happening, the second version would simply return the whole string, whilst the first would raise an exception, which is exactly the intent here.

Reply:
Agreed, but you have a test `if ':' not in decoded` just above, so the split will always happen. If you prefer the tuple-destructuring syntax, maybe one could use the EAFP pattern (mentioned in [our brand-new Best Practices document](https://github.com/scality/metalk8s/blob/development/2.0/docs/best_practices/python.md#dont-write-code-vulnerable-to-time-of-check-to-time-of-use)), something like:

```python
try:
    token_username, _ = decoded.split(':', 1)
except ValueError:
    log.warning('Invalid Basic token format: missing ":"')
    return False
```